x-pack/solutions/security/plugins/security_solution/public/data_view_picker/redux/selectors.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { createSelector } from '@reduxjs/toolkit';
+
+import type { DataViewPickerScopeName } from '../constants';
+import type { RootState } from './reducer';
+
+export const sourcererAdapterSelector = (scope: DataViewPickerScopeName) =>
+  createSelector([(state: RootState) => state.dataViewPicker], (dataViewPicker) => {
+    const scopedState = dataViewPicker[scope];
+
+    return {
+      ...scopedState,
+      dataView: scopedState.dataView ? scopedState.dataView : { title: '', id: '' },
+      indicesExist: !!dataViewPicker[scope]?.dataView?.title?.split(',')?.length,
```

Comment:
Is the plan in the future to replace this with the presence of the dataview itself, or keep the indices exist abstraction?? This could be a good use case for a `withDataView` wrapper...but it doesn't create the dataview in the wrapper...it just provides a fall back component if no dataview actually exists (even if that's just the landing page)

Reply:
No idea, depends if we actually need this anywhere in its current form.